### PR TITLE
DAOS-10147 EC: clear EC stripe list if EC aggregation is aborted

### PR DESCRIPTION
**Pull request by wangdi1**
clear EC stripe list if EC aggregation is aborted, since post_cb
might be skipped.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/object/srv_ec_aggregate.c [hunk 1]
```diff
@@ -2319,7 +2321,9 @@ agg_iterate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 		if (!(*acts & VOS_ITER_CB_SKIP))
 			agg_reset_pos(type, agg_entry);
 		if (ec_aggregate_yield(agg_param)) {
+			/* post_cb may not be called, so let's process current stripe */
 			D_DEBUG(DB_EPC, "EC aggregation aborted\n");
+			agg_process_curr_stripe(ih, agg_param, entry, agg_entry, acts);
```
Comment (jolivier23) [inline src/object/srv_ec_aggregate.c:2326]: I am not sure this is safe to do after re-probe.   Is there not an alternative where we simply clean-up what we've allocated?   Or is it too late at this point.    I believe the nested iterator will do a dbtree_iter_fetch on the entry in question and I'm not sure it's valid.  If more I/O happened, the tree pointers may have moved.  The alternative is to add a callback for "exit" but that probably isn't the right answer for 2.0.

**Review by jolivier23 — CHANGES_REQUESTED** [review]

**Review by jolivier23 — APPROVED** [review]

File: src/object/srv_ec_aggregate.c [hunk 2]
```diff
@@ -2307,6 +2306,7 @@ agg_iterate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	}
 
 	if (rc < 0) {
+		agg_clear_extents(agg_entry);
```
Comment (NiuYawei) [inline src/object/srv_ec_aggregate.c:2309]: Shouldn't such cleanup function called in cont_ec_aggregate_cb()? what if vos_iterate() abort on other errors (error not returned by pre callback)?

**Review by NiuYawei — CHANGES_REQUESTED** [review]

**Review by NiuYawei — APPROVED** [review]

**Comment by johannlombardi** [comment]
This needs to run with the test tag suggested by Phil:
Test-tag: largefilecount

**Comment by johannlombardi** [comment]
This needs to run with the test tag suggested by Phil:
Test-tag: largefilecount

**Review by jgmoore-or — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**